### PR TITLE
integration_tests: add checks to takeoff/land test

### PR DESCRIPTION
**Pull request by julianoes**
This adds some checks to the ActionSimpleHover tests to actually check
what PX4 SITL is executing rather than just checking if we get the
correct responses.

This could be an example for CI tests for PX4 using DroneCore.

@dagar have a look

To test this:
- Start PX4 SITL
- Then start the test:
    ```
    make && build/default/integration_tests_runner --gtest_filter="SitlTest.ActionSimpleHover*"
    ```
    
Or if you have Firmware and DroneCore next to each other in a folder:

```
make && AUTOSTART_SITL=1 build/default/integration_tests_runner --gtest_filter="SitlTest.ActionSimpleHover*"
```

### CODE REVIEW

File: integration_tests/simple_hover.cpp [hunk 1]
```diff
@@ -13,31 +33,55 @@ TEST_F(SitlTest, ActionSimpleHover)
 
     // Wait for device to connect via heartbeat.
     std::this_thread::sleep_for(std::chrono::seconds(2));
+    ASSERT_TRUE(dc.is_connected());
 
     Device &device = dc.device();
 
-    while (!device.telemetry().health_all_ok()) {
+    for (int i = 0; ; ++i) {
```
Comment (hamishwillee) [inline integration_tests/simple_hover.cpp]: Just FMI, why not a while loop? Seem odd to see an infinite for loop :-)

Comment (hamishwillee) [reply integration_tests/simple_hover.cpp]: Ignore that, I see you want to count out 10 seconds.

Comment (julianoes) [reply integration_tests/simple_hover.cpp]: Yer it's a bit ugly but I haven't found a prettier way yet.

An alternative is to use the async interface and then do a `wait_for` on the promise.

Comment (hamishwillee) [reply integration_tests/simple_hover.cpp]: Not prettier, but perhaps a little more intuitive
```
int itter=0;
while (!device.telemetry().health_all_ok()) {
   std::cout << "waiting for device to be ready" << std::endl;
   std::this_thread::sleep_for(std::chrono::seconds(1));
   ASSERT_LT(++itter, 10);
}
```

Comment (julianoes) [reply integration_tests/simple_hover.cpp]: Yer, you're right. That's nicer. I'll change it.

**Comment by julianoes** [comment]
Replaced by #110.

**Pull request closed without merging**